### DISCUSSION TITLE
unstretched image scaling feature request

### DISCUSSION THREAD

**Discussion by wb1016** [requarks/wiki#7362] [Help / Questions] [answered]
currently, the wiki.js requires both width and height for image scaling. this is quite devastating because there are many devices have various DPI and resolution. if the given image scale is bigger than screen resolution, the image is displayed as stretched.(see example below)
![image](https://github.com/user-attachments/assets/5c800435-d734-4146-bbb7-51a80839f220)
to solve this problem, we need to fix the image aspect ratio. when given image scale is bigger than screen resolution, the image height should be scaled relative to content width. this can be achieved by giving only width to image scale factor. ffmpeg has good example. they accept format like this:
```-vf scale=1920:-1```
the `-1` means automatic scaling.
please add this feature to wiki.js. it will be good to have.

**Comment by fnafox** [comment, marked as answer]
I'm not sure I understand what you mean. Images get scaled down to available screen size by default.
If you want to specify the size of the embedded image, you can do so like this.
Omitting the height value will keep the aspect ratio.
```md
<!-- scaling an image to an exact size -->
![image.png](/res/image.png =1920x1080)

<!-- scaling image using relative height -->
![image.png](/res/image.png =1920x)
```

**Reply by wb1016** [reply]
wow. i didnt know that. thanks.
